src/cmd/usb-util.js

```diff
@@ -15,6 +16,55 @@ const REOPEN_TIMEOUT = 60000;
 // When reopening a device that was about to reset, give it some time to boot into the firmware
 const REOPEN_DELAY = 500;
 
+
+async function _getDeviceId(device) {
+	let id = null;
+	try {
+		await device.open();
+		return device._id;
+	} catch (err) {
+		// ignore error
+	} finally {
+		if (device.isOpen) {
+			await device.close();
+		}
+	}
+	return id;
+}
+
+async function _getDeviceMode(device) {
+	if (device.isInDfuMode) {
+		return 'DFU';
+	}
+
+	try {
+		await device.open();
+		const mode = await device.getDeviceMode({ timeout: 10 * 1000 });
+		if (mode && (mode !== 'NORMAL')) {
+			return mode;
+		}
+	} catch (err) {
+		if (err instanceof TimeoutError) {
+			return 'UNKNOWN';
+		} else {
+			throw new Error(`Unable to get device mode: ${err.message}`);
+		}
+	} finally {
+		if (device.isOpen) {
+			await device.close();
+		}
+	}
+}
+
+async function _getDeviceName({ id, api, auth, ui }) {
+	try {
+		const device = await getDevice({ id, api, auth, ui });
+		return device.name;
+	} catch (err) {
+		// ignore error
```

Comment:
Can you see if you can distinguish the 2 cases in this error handler and return the following names:
`<no name>`: the device hasn't been named yet
`<unknown>`: if the API request failed (errored with 404 Not Found because you don't have access to the device or if the API request couldn't be performed because the computer is offline and the device attributes are not in cache)

`particle usb list` does show `<no name>`
```
$ particle usb list
<no name> [18002b000247353138383138] (Photon, LISTENING)
extra_p2 [0a10aced202194944a02c5f4] (Photon 2 / P2)
```

Reply:
yes - let me try that

Reply:
Hope this looks correct - added in the change
If connection is lost
```
? Which device would you like to select? (Use arrow keys)
❯ <unknown> [0a10aced202194944a02c4d4] (Photon 2 / P2, DFU) 
  <unknown> [e00fce6819ef5f971ea9563a] (Argon) 
```
No name
```
? Which device would you like to select? (Use arrow keys)
❯ seal_penguin [0a10aced202194944a02c4d4] (Photon 2 / P2, DFU) 
  <no name> [e00fce6819ef5f971ea9563a] (Argon) 
```